### DISCUSSION TITLE
Access previous data in event trigger?

### DISCUSSION THREAD

**Discussion by llamafilm** [netbox-community/netbox#14809] [Q&A]
I’m interested in building an event trigger that will run when a device is updated, to push the new data to my Prometheus monitoring system. But I only want to trigger when certain fields are modified on a device, such as name, primary IP, or status. For example I don’t want it to run if only the rack position was changed. Is this possible? I could certainly build the logic into the script, but I need a way to access the previous device properties to compare with current.

**Comment by markkuleinio** [comment]
> I could certainly build the logic into the script, but I need a way to access the previous device properties to compare with current.

See https://docs.netbox.dev/en/stable/integrations/webhooks/#available-context, there it shows the snapshot data included in webhooks. I'd assume something similar is also included when using scripts instead of webhooks.

**Reply by llamafilm** [reply]
Thanks for the idea.  I looked around there but didn't see anything similar for scripts.  The script receives a `data` object which is the same as for a webhook, but I don't see a way to get the `snapshots` or other objects.

**Reply by markkuleinio** [reply]
You are correct, there are no snapshots in the script data. You may want to open a feature request for that.

**Reply by DanSheps** [reply]
You are correct, it does not include the "snapshots" parameter

**Reply by llamafilm** [reply]
I like the idea of adding snapshots to the script trigger, especially since it's already there in the web hook.

**Reply by llamafilm** [reply]
I opened FR #14896

**Comment by DanSheps** [comment]
You have two options:

1. Craft conditions such that it pushes only when needed
2. Write a custom plugin that listens for a signal on the desired model and fires off

**Reply by llamafilm** [reply]
From what I can see, conditions only have access to the current state.  I would like is to compare a few key fields (status, primary_ip, name) of the current state against the previous state.

**Reply by markkuleinio** [reply]
Third option: don't use scripts but use webhook and NetBox REST API as needed. That's what I do exclusively. This is my old but still useful template for having a webhook receiver on the localhost: https://majornetwork.net/2020/10/webhook-listener-with-fastapi/
